### DISCUSSION TITLE
Memory Error on Laravel Forge vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php

### DISCUSSION THREAD

**Discussion by daylightstudio** [statamic/cms#5095] [Troubleshooting] [answered]
I keep getting 
```
allowed memory size of 1073741824 bytes exhausted (tried to allocate 8192 bytes) {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Allowed memory size of 1073741824 bytes exhausted (tried to allocate 8192 bytes) at /home/forge/xxx/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php:78)
```
for pages that don't exist (e.g. a page that should be showing a 404 error page). It works fine locally but on Laravel Forge it's producing the error. I've seen the error elsewhere on non-404 error pages too. I thought the server may need some upgrading but it seems like it should be adequate and the PHP memory is set all the way to 1024MB which should be more than enough. Viewing server memory available also looks to be adequate. Any thoughts? Is it an nginx configuration issue?

**Comment by duncanmcclean** [comment]
Hah, I had this exact same issue yesterday afternoon.

Are you by any chance using the Redirects addon?

**Comment by daylightstudio** [comment]
Why yes I am.

**Reply by duncanmcclean** [reply]
Set this to `false`, it'll fix your 404 problems: https://github.com/riasvdv/statamic-redirect/blob/main/config/redirect.php#L18

**Comment by daylightstudio** [comment, marked as answer]
Thanks for the tip @duncanmcclean!

(future readers be sure to publish the config if you haven't already `php artisan vendor:publish --tag=statamic-redirect-config`).

I'm no longer getting memory errors. However, I was still getting max execution time errors after 60 seconds:
```
Maximum execution time of 60 seconds exceeded {"userId":"xxx","exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Maximum execution time of 60 seconds exceeded at /home/forge/xxx/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php:271)
```
The solution for that was to simply remove all the files in the `storage/redirect/errors` folder.
